addons/website_mass_mailing/static/src/js/website_mass_mailing.js

```diff
@@ -71,50 +74,83 @@ publicWidget.registry.newsletter_popup = publicWidget.Widget.extend({
     disabledInEditableMode: false,
 
+    read_events: {
+        'click .o_popup_subscribe_btn': '_onClickSubscribe',
+    },
+    xmlDependencies: ['/website_mass_mailing/static/src/xml/website_mass_mailing.xml'],
+
+    /**
+     * @override
+     */
     start: function () {
         var self = this;
+        var defs = [this._super.apply(this, arguments)];
+        this.websiteID = this._getContext().website_id;
+        this.listID = this.$target.attr('data-list-id');
 
-        // Compatibility: rebuilding a correct modal from user database
-        // content. Since the first version, the modal is saved in databases
-        // directly but has never used a correct structure. While it was working
-        // with BS3, it is not with BS4.
-        // TODO review the newsletter popup creation, save and loading.
-        var $modal = this.$('.modal');
-        if ($modal.is('.modal-dialog')) {
-            $modal.removeClass('modal-md modal-dialog');
-            var $modalContent = $modal.find('.modal-content');
-            $modalContent.wrapAll($('<div/>', {class: 'modal-dialog'}));
-        }
-
-        var popupcontent = self.$target.find(".o_popup_content_dev").empty();
-        if (!self.$target.data('list-id')) {
-            return;
+        if (!this.listID || (utils.get_cookie(_.str.sprintf("newsletter-popup-%s-%s", this.listID, this.websiteID)) && !self.editableMode)) {
+            return Promise.all(defs);
         }
 
-        this._rpc({
+        defs.push(this._rpc({
             route: '/website_mass_mailing/get_content',
             params: {
-                newsletter_id: self.$target.data('list-id'),
+                newsletter_id: self.listID,
             },
         }).then(function (data) {
-            if (data.content) {
-                $('<div></div>').append(data.content).appendTo(popupcontent);
-            }
-            self.$target.find('input.popup_subscribe_email').val(data.email || "");
-            self.redirect_url = data.redirect_url;
+            self.$target.find('.o_newsletter_modal').remove();
+            self.$target.append(qweb.render('website_mass_mailing.popup', {
+                content: data.popup_content,
+                isEditMode: self.editableMode,
+            }));
+            self.$target.find('.o_mm_content_block')
+                .append(qweb.render('website_mass_mailing.subscribe_input'));
+
             if (!self.editableMode && !data.is_subscriber) {
-                $(document).on('mouseleave', _.bind(self.show_banner, self));
+                self.$target.find('input.o_popup_subscribe_email')
+                    .val(data.email ? data.email : "");
+                if (config.device.isMobile) {
+                    setTimeout(function () {
+                        self._showBanner();
+                    }, 5000);
+                } else {
+                    $(document).on('mouseleave.open_popup_event', _.bind(self._showBanner, self));
```

Comment:
I know it was like that before but it is a good time to replace the use of old `_.bind`, use `self._showBanner.bind(self)` directly